Index: src/Gscore/WsApp.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Gscore/WsApp.php	(revision 0acf5d5d49887d4901ccf01adf2afcd6ba896deb)
+++ src/Gscore/WsApp.php	(revision )
@@ -12,22 +12,19 @@
     public function __construct( $config ) {
         $this->config = $config; //may be need in future
         $this->clients = new \SplObjectStorage;
-
     }
 
-    public function onOpen(ConnectionInterface $conn) {
+    public function onOpen(ConnectionInterface $conn) {}
 
-        $conn->username = 'guest';
-        $this->clients->attach($conn);
-
-        echo "New connection! ({$conn->resourceId})" . PHP_EOL;
-        echo $this->getOnline();
-
-    }
-
     public function onMessage(ConnectionInterface $client, $msg) {
-        $response = $this->loginAction( $client, $msg );
+        $client->username = $msg;
+        $this->clients->attach($client);
+        $response = new JsonMessage('login', array(
+            'status' => 200,
+            'resource_id' => $client->resourceId
+            ));                                                                   
         Sender::send($client , $response);
+        echo $this->getOnline();
     }
 
     public function onClose(ConnectionInterface $conn) {
@@ -43,14 +40,14 @@
         $conn->close();
     }
 
-    public function onZMQMessage( $zmqMessage ){
+    public function onZMQMessage( $zmqMessage ) {
         $entry = json_decode($zmqMessage , true);
         $userNames = $entry['usernames'];
         $message = JsonMessage::createFromArray($entry['message']);
         echo "onZMQ message: start to send clients" . PHP_EOL;
         $time1 = time();
 
-        if ($userNames === array()){
+        if ($userNames === array()) {
             foreach ($this->clients as $client) {
                 Sender::send($client, $message);
             }
@@ -64,7 +61,7 @@
         echo "onZMQ message: sended to clients in ". (time() - $time1) . 'mc'. PHP_EOL;
     }
 
-    protected function getOnline(){
+    protected function getOnline() {
         $result = '';
         foreach ($this->clients as $client){
             $result .= self::getConnectionInfo($client);
@@ -72,12 +69,11 @@
         return 'ONLINE: '. $result . PHP_EOL ;
     }
 
-    protected static function getConnectionInfo( $client ){
+    protected static function getConnectionInfo( $client ) {
         return "{$client->username}({$client->resourceId})";
     }
 
-    protected function loginAction(ConnectionInterface $from, $msg){
+    protected function loginAction(ConnectionInterface $from, $msg) {
-
         $wsseValidator = new WsseValidator();
         if ($wsseValidator->isValid($msg)){
             $from->username = $wsseValidator->getUsername();
@@ -91,5 +87,4 @@
             return new JsonMessage('login', array('status' => 401));
         }
     }
-
 }
\ No newline at end of file
